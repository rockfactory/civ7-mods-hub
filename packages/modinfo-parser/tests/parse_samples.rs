use modinfo_parser::Mod;

#[test]
fn parse_samples() {
    insta::glob!("samples/*.modinfo", |path| {
        let text = std::fs::read_to_string(path).unwrap();
        let parsed = Mod::parse(text.as_bytes()).unwrap();
        insta::with_settings!({ sort_maps => true }, {
            insta::assert_debug_snapshot!(parsed);
            let serialized = parsed.to_string().unwrap();
            let reparsed = Mod::parse(serialized.as_bytes()).unwrap();
            assert_eq!(parsed, reparsed);
        });
    });
}
