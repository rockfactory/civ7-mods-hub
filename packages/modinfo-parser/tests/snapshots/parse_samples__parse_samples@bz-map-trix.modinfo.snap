---
source: packages/modinfo-parser/tests/parse_samples.rs
expression: parsed
input_file: packages/modinfo-parser/tests/samples/bz-map-trix.modinfo
---
Mod {
    id: "bz-map-trix",
    version: Some(
        "10801",
    ),
    properties: Properties {
        name: Some(
            "LOC_MOD_BZ_MAP_TRIX_NAME",
        ),
        description: Some(
            "LOC_MOD_BZ_MAP_TRIX_DESCRIPTION",
        ),
        authors: Some(
            "beezany",
        ),
        package: Some(
            "Mod",
        ),
        affects_saved_games: false,
        special_thanks: None,
        version: Some(
            "1.8.1",
        ),
        compatibility: None,
        url: Some(
            "https://forums.civfanatics.com/resources/map-trix.31950/",
        ),
    },
    dependencies: [
        Dependency {
            id: "base-standard",
            title: Some(
                "LOC_MODULE_BASE_STANDARD_NAME",
            ),
        },
    ],
    localized_text: [
        "text/en_us/ModInfoText.xml",
        "text/de_DE/ModInfoText.xml",
        "text/it_IT/ModInfoText.xml",
        "text/ko_KR/ModInfoText.xml",
    ],
    action_criteria: [
        ActionCriterion {
            id: "always",
            always_met: true,
            age_in_use: None,
            mod_in_use: None,
        },
    ],
    action_groups: [
        ActionGroup {
            id: "bz-map-trix-menu",
            scope: Some(
                "shell",
            ),
            criteria: Some(
                "always",
            ),
            properties: ActionGroupProperties {
                load_order: Some(
                    1000,
                ),
            },
            actions: [
                ActionGroupActions {
                    import_files: [],
                    ui_scripts: [
                        ActionItem {
                            path: "ui/options/bz-map-trix-options.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/options/mod-options-decorator.js",
                            locale: None,
                        },
                    ],
                    update_database: [],
                    update_text: [
                        ActionItem {
                            path: "text/en_us/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/de_DE/InGameText.xml",
                            locale: Some(
                                "de_DE",
                            ),
                        },
                        ActionItem {
                            path: "text/it_IT/InGameText.xml",
                            locale: Some(
                                "it_IT",
                            ),
                        },
                        ActionItem {
                            path: "text/ko_kr/InGameText.xml",
                            locale: Some(
                                "ko_KR",
                            ),
                        },
                        ActionItem {
                            path: "text/ru_RU/InGameText.xml",
                            locale: Some(
                                "ru_RU",
                            ),
                        },
                    ],
                },
            ],
        },
        ActionGroup {
            id: "bz-map-trix-ui",
            scope: Some(
                "game",
            ),
            criteria: Some(
                "always",
            ),
            properties: ActionGroupProperties {
                load_order: Some(
                    1000,
                ),
            },
            actions: [
                ActionGroupActions {
                    import_files: [],
                    ui_scripts: [
                        ActionItem {
                            path: "ui/diplo-ribbon/bz-panel-yield-banner.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/options/bz-map-trix-options.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/options/mod-options-decorator.js",
                            locale: None,
                        },
                        ActionItem {
                            path: "ui/tooltips/bz-plot-tooltip.js",
                            locale: None,
                        },
                    ],
                    update_database: [],
                    update_text: [
                        ActionItem {
                            path: "text/en_us/InGameText.xml",
                            locale: None,
                        },
                        ActionItem {
                            path: "text/de_DE/InGameText.xml",
                            locale: Some(
                                "de_DE",
                            ),
                        },
                        ActionItem {
                            path: "text/it_IT/InGameText.xml",
                            locale: Some(
                                "it_IT",
                            ),
                        },
                        ActionItem {
                            path: "text/ko_kr/InGameText.xml",
                            locale: Some(
                                "ko_KR",
                            ),
                        },
                        ActionItem {
                            path: "text/ru_RU/InGameText.xml",
                            locale: Some(
                                "ru_RU",
                            ),
                        },
                    ],
                },
            ],
        },
    ],
}
